odl/discr/grid.py

```diff
@@ -1190,33 +1020,29 @@ def uniform_sampling(intv_prod, num_nodes, as_midp=True):
         Number of nodes per axis. For dimension >= 2, a `tuple`
         is required. All entries must be positive. Entries
         corresponding to degenerate axes must be equal to 1.
-    as_midp : `bool`, optional
-        If `True`, the midpoints of an interval partition will be
-        returned, which excludes the endpoints. Otherwise,
-        equispaced nodes including the endpoints are generated.
-        Note that the resulting strides are different.
-        Default: `True`.
 
     Returns
     -------
     sampling : `RegularGrid`
         Uniform sampling grid for the interval product
 
+    See also
+    --------
+    uniform_partition :
+        divide interval products into equally sized subsets
+
     Examples
     --------
     >>> from odl import IntervalProd
     >>> rbox = IntervalProd([-1.5, 2], [-0.5, 3])
-    >>> grid = uniform_sampling(rbox, [2, 5])
-    >>> grid.coord_vectors
-    (array([-1.25, -0.75]), array([ 2.1,  2.3,  2.5,  2.7,  2.9]))
-    >>> grid = uniform_sampling(rbox, [2, 5], as_midp=False)
+    >>> grid = uniform_sampling(rbox, [3, 3])
     >>> grid.coord_vectors
-    (array([-1.5, -0.5]), array([ 2.  ,  2.25,  2.5 ,  2.75,  3.  ]))
+    (array([-1.5, -1. , -0.5]), array([ 2. ,  2.5,  3. ]))
     """
     num_nodes = np.atleast_1d(num_nodes).astype('int64')
```

Comment:
This allows rounding, i.e. num_nodes=[1.5, 3.2], I'd add `casting='safe'` so it requires integer input.